### PR TITLE
Improve mongodb connector performance

### PR DESCRIPTION
**Pull request by SergeTupchiy**
Fixes  EMQX-11825
Release version: e5.5.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by thalesmg — APPROVED** [review]
Maybe just add a `perf` changelog entry?

File: changes/ee/perf-12494.en.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Improve MongoDB connector performance.
+
+- improve mongodb-erlang client performance
+  [mongodb-erlang PR](https://github.com/emqx/mongodb-erlang/pull/41)
+- un-deprecate Mongodb `topology.pool_size` configuration parameter and set it to 10 by default.
```
Comment (thalesmg) [inline changes/ee/perf-12494.en.md:5]: Maybe we could leave this out, since it's still hidden (undocumented) and only a lever for tuning if needed.

**Review by thalesmg — APPROVED** [review]

File: changes/ee/perf-12494.en.md (hunk 1 above)
Comment (SergeTupchiy) [reply changes/ee/perf-12494.en.md]: Makes sense, removed.

**Review by thalesmg — APPROVED** [review]
:beers:

**Pull request merged**